File: ios/Sources/SportsGreenMooveApp/PublishControls.swift
import SwiftUI

struct PublishDriverRequiredCard: View {
    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            Text("MODE CONDUCTEUR REQUIS")
                .font(.sgmDisplay(20))
                .tracking(.sgmWide(for: 20))
                .foregroundStyle(SGM.textPrimary)
            Text("Passez en rôle conducteur depuis le profil pour publier un trajet vérifié.")
                .font(.sgmBody(14, weight: .medium))
                .foregroundStyle(SGM.textSecondary)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(18)
        .background(SGM.bgCard, in: RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous).stroke(SGM.border, lineWidth: 1))
    }
}

struct PublishStepper: View {
    let step: Int

    var body: some View {
        HStack(spacing: 8) {
            ForEach(1...3, id: \.self) { item in
                Text("\(item)")
                    .font(.sgmBody(12, weight: .bold))
                    .foregroundStyle(item <= step ? SGM.textOnGreen : SGM.textMuted)
                    .frame(width: 28, height: 28)
                    .background(item <= step ? SGM.green : SGM.bgCard, in: Circle())
                    .overlay(Circle().stroke(SGM.border, lineWidth: 1))
                if item < 3 {
                    Capsule().fill(item < step ? SGM.green : SGM.border).frame(height: 3)
                }
            }
        }
        .padding(.horizontal, SGMSpace.padScreen)
        .padding(.bottom, 6)
    }
}

struct PublishTitle: View {
    let text: String

    init(_ text: String) {
        self.text = text
    }

    var body: some View {
        Text(text)
            .font(.sgmDisplay(20))
            .tracking(.sgmWide(for: 20))
            .foregroundStyle(SGM.textPrimary)
            .frame(maxWidth: .infinity, alignment: .leading)
    }
}

struct PublishChoiceRow: View {
    let label: String
    let values: [String]
    @Binding var selected: String

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(label)
                .font(.sgmBody(13, weight: .semibold))
                .foregroundStyle(SGM.textSecondary)
            ScrollView(.horizontal, showsIndicators: false) {
                HStack(spacing: 6) {
                    ForEach(values, id: \.self) { value in
                        SGMChip(text: value, selected: selected == value) {
                            selected = value
                        }
                        .frame(width: value.count > 8 ? 126 : 92)
                    }
                }
            }
        }
        .publishRowStyle()
    }
}

struct PublishTextRow: View {
    let icon: SGMIcon
    let placeholder: String
    @Binding var text: String

    var body: some View {
        HStack(spacing: 10) {
            SGMIconView(icon: icon, size: 18, color: SGM.green)
            TextField(placeholder, text: $text)
                .font(.sgmBody(14, weight: .medium))
                .foregroundStyle(SGM.textPrimary)
        }
        .publishRowStyle()
    }
}

struct PublishToggleRow: View {
    let label: String
    @Binding var selected: Bool

    var body: some View {
        HStack {
            Text(label)
                .font(.sgmBody(13, weight: .semibold))
                .foregroundStyle(SGM.textSecondary)
            Spacer()
            Button { selected.toggle() } label: {
                Capsule()
                    .fill(selected ? SGM.green : SGM.bgInput)
                    .frame(width: 44, height: 24)
                    .overlay(alignment: selected ? .trailing : .leading) {
                        Circle()
                            .fill(SGM.textOnGreen)
                            .frame(width: 18, height: 18)
                            .padding(3)
                    }
            }
            .buttonStyle(.plain)
        }
        .publishRowStyle()
    }
}

struct PublishSeatsRow: View {
    @Binding var seats: Int

    var body: some View {
        HStack {
            Text("Places disponibles")
                .font(.sgmBody(13, weight: .semibold))
                .foregroundStyle(SGM.textSecondary)
            Spacer()
            SGMCircleIconButton(icon: .chevronLeft, size: 28) {
                seats = max(1, seats - 1)
            }
            Text("\(seats)")
                .font(.sgmDisplay(22))
                .foregroundStyle(SGM.green)
                .frame(width: 24)
            SGMCircleIconButton(icon: .plus, size: 28) {
                seats = min(6, seats + 1)
            }
        }
        .publishRowStyle()
    }
}

private struct PublishRowStyle: ViewModifier {
    func body(content: Content) -> some View {
        content
            .padding(.horizontal, 14)
            .padding(.vertical, 12)
            .background(SGM.bgSurface, in: RoundedRectangle(cornerRadius: SGMRadius.md, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: SGMRadius.md, style: .continuous).stroke(SGM.border, lineWidth: 1))
    }
}

private extension View {
    func publishRowStyle() -> some View {
        modifier(PublishRowStyle())
    }
}

File: android/app/src/main/java/be/sportgreenmoove/app/ui/PublishControls.kt
package be.sportgreenmoove.app.ui

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.RowScope
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicTextField
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.em
import androidx.compose.ui.unit.sp
import be.sportgreenmoove.app.design.Sgm
import be.sportgreenmoove.app.design.SgmColor
import be.sportgreenmoove.app.design.SgmRadius
import be.sportgreenmoove.app.design.SgmType

@Composable
fun PublishInput(icon: SgmIcon, placeholder: String, value: String, onValue: (String) -> Unit) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(SgmRadius.MD))
            .background(Sgm.colors.bgInput)
            .border(BorderStroke(1.5.dp, Sgm.colors.border), RoundedCornerShape(SgmRadius.MD))
            .padding(horizontal = 14.dp, vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        SgmLineIcon(icon, tint = SgmColor.Green, modifier = Modifier.size(18.dp))
        Box(Modifier.weight(1f)) {
            if (value.isEmpty()) Text(placeholder, style = PublishBodyStyle(Sgm.colors.textMuted))
            BasicTextField(value = value, onValueChange = onValue, textStyle = PublishBodyStyle(Sgm.colors.textPrimary), singleLine = true, modifier = Modifier.fillMaxWidth())
        }
    }
}

@Composable
fun PublishChoiceRow(label: String, values: List<String>, selected: String, onSelect: (String) -> Unit, titleFirst: Boolean = false) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(SgmRadius.MD))
            .background(Sgm.colors.bgSurface)
            .border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(SgmRadius.MD))
            .padding(horizontal = 14.dp, vertical = 10.dp),
        verticalArrangement = Arrangement.spacedBy(if (titleFirst) 8.dp else 0.dp),
    ) {
        Text(label, style = SgmType.BodySM.copy(color = Sgm.colors.textSecondary, fontSize = 13.sp, fontWeight = FontWeight.SemiBold))
        Row(modifier = Modifier.horizontalScroll(rememberScrollState()).padding(top = if (titleFirst) 0.dp else 8.dp), horizontalArrangement = Arrangement.spacedBy(6.dp)) {
            values.forEach { value -> V2Chip(value, selected = selected == value, onClick = { onSelect(value) }) }
        }
    }
}

@Composable
fun PublishToggleRow(label: String, selected: Boolean, onClick: () -> Unit) {
    PublishControlRow(label) {
        Box(
            modifier = Modifier
                .size(width = 44.dp, height = 24.dp)
                .clip(RoundedCornerShape(999.dp))
                .background(if (selected) SgmColor.Green else Sgm.colors.bgInput)
                .border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(999.dp))
                .clickable(onClick = onClick),
        ) {
            Box(Modifier.padding(start = if (selected) 22.dp else 2.dp, top = 2.dp).size(18.dp).clip(CircleShape).background(SgmColor.TextOnGreen))
        }
    }
}

@Composable
fun PublishSeatsRow(seats: Int, onSeats: (Int) -> Unit) {
    PublishControlRow("Places disponibles") {
        V2CircleIconButton(SgmIcon.ChevronLeft, onClick = { onSeats(seats - 1) }, size = 28)
        Text(seats.toString(), style = SgmType.DisplayXL.copy(color = SgmColor.Green, fontSize = 22.sp), modifier = Modifier.size(width = 24.dp, height = 28.dp), textAlign = TextAlign.Center)
        V2CircleIconButton(SgmIcon.Plus, onClick = { onSeats(seats + 1) }, size = 28)
    }
}

@Composable
fun PublishPriceRow(price: String, onPrice: (String) -> Unit) {
    PublishControlRow("Prix estimé") {
        BasicTextField(value = price, onValueChange = onPrice, textStyle = PublishBodyStyle(Sgm.colors.textPrimary).copy(textAlign = TextAlign.End), singleLine = true, modifier = Modifier.size(width = 70.dp, height = 28.dp).clip(RoundedCornerShape(8.dp)).background(Sgm.colors.bgInput).border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(8.dp)).padding(horizontal = 10.dp, vertical = 6.dp))
        Text("€", style = SgmType.BodySM.copy(color = Sgm.colors.textSecondary, fontSize = 14.sp, fontWeight = FontWeight.Bold))
    }
}

@Composable
fun PublishDriverRequired() {
    Column(modifier = Modifier.fillMaxWidth().clip(RoundedCornerShape(SgmRadius.LG)).background(Sgm.colors.bgCard).border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(SgmRadius.LG)).padding(18.dp), verticalArrangement = Arrangement.spacedBy(10.dp)) {
        Text("MODE CONDUCTEUR REQUIS", style = SgmType.DisplayLG.copy(color = Sgm.colors.textPrimary, fontSize = 20.sp, letterSpacing = 0.06.em))
        Text("Passez en rôle conducteur depuis le profil pour publier un trajet vérifié.", style = SgmType.BodySM.copy(color = Sgm.colors.textSecondary, lineHeight = 20.sp))
    }
}

@Composable
fun PublishTitle(text: String) = Text(text, style = SgmType.DisplayLG.copy(color = Sgm.colors.textPrimary, fontSize = 20.sp, letterSpacing = 0.06.em), modifier = Modifier.padding(bottom = 4.dp))

@Composable
fun PublishStepDot(value: Int, active: Boolean) {
    Box(Modifier.size(28.dp).clip(CircleShape).background(if (active) SgmColor.Green else Sgm.colors.bgCard).then(if (active) Modifier else Modifier.border(BorderStroke(1.dp, Sgm.colors.border), CircleShape)), contentAlignment = Alignment.Center) {
        Text(value.toString(), style = SgmType.DisplayLG.copy(color = if (active) SgmColor.TextOnGreen else Sgm.colors.textMuted, fontSize = 14.sp))
    }
}

@Composable
fun PublishSummaryRow(label: String, value: String) {
    Row(Modifier.fillMaxWidth()) {
        Text(label, style = SgmType.BodySM.copy(color = Sgm.colors.textMuted, fontSize = 13.sp, fontWeight = FontWeight.Medium), modifier = Modifier.weight(1f))
        Text(value, style = SgmType.BodySM.copy(color = Sgm.colors.textPrimary, fontSize = 13.sp, fontWeight = FontWeight.Bold))
    }
}

@Composable
fun PublishBodyStyle(color: Color) = SgmType.BodySM.copy(color = color, fontSize = 14.sp, fontWeight = FontWeight.Medium)

@Composable
private fun PublishControlRow(label: String, trailing: @Composable RowScope.() -> Unit) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(SgmRadius.MD))
            .background(Sgm.colors.bgSurface)
            .border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(SgmRadius.MD))
            .padding(horizontal = 14.dp, vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Text(label, style = SgmType.BodySM.copy(color = Sgm.colors.textSecondary, fontSize = 13.sp, fontWeight = FontWeight.SemiBold), modifier = Modifier.weight(1f))
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(6.dp), content = trailing)
    }
}
